plugins/crypto/mbedtls/certificategroup.c

```diff
@@ -719,6 +720,175 @@ UA_CertificateUtils_getKeySize(UA_ByteString *certificate,
     return UA_STATUSCODE_GOOD;
 }
 
+UA_StatusCode
+UA_CertificateUtils_comparePublicKeys(const UA_ByteString *certificate1,
+                                      const UA_ByteString *certificate2) {
+    UA_StatusCode retval = UA_STATUSCODE_GOOD;
+
+    mbedtls_x509_crt cert1;
+    mbedtls_x509_crt cert2;
+    mbedtls_x509_csr csr1;
+    mbedtls_x509_csr csr2;
+    mbedtls_mpi N1, E1;
+    mbedtls_mpi N2, E2;
+
+    UA_ByteString data1 = UA_mbedTLS_CopyDataFormatAware(certificate1);
+    UA_ByteString data2 = UA_mbedTLS_CopyDataFormatAware(certificate2);
+
+    mbedtls_x509_crt_init(&cert1);
+    mbedtls_x509_crt_init(&cert2);
+    mbedtls_x509_csr_init(&csr1);
+    mbedtls_x509_csr_init(&csr2);
+    mbedtls_mpi_init(&N1);
+    mbedtls_mpi_init(&E1);
+    mbedtls_mpi_init(&N2);
+    mbedtls_mpi_init(&E2);
+
+    int mbedErr = mbedtls_x509_crt_parse(&cert1, data1.data, data1.length);
+    if(mbedErr) {
+        /* Try to load as a csr */
+        mbedErr = mbedtls_x509_csr_parse(&csr1, data1.data, data1.length);
+        if(mbedErr) {
+            retval = UA_STATUSCODE_BADINTERNALERROR;
+            goto cleanup;
+        }
+    }
+
+    mbedErr = mbedtls_x509_crt_parse(&cert2, data2.data, data2.length);
+    if(mbedErr) {
+        /* Try to load as a csr */
+        mbedErr = mbedtls_x509_csr_parse(&csr2, data2.data, data2.length);
+        if(mbedErr) {
+            retval = UA_STATUSCODE_BADINTERNALERROR;
+            goto cleanup;
+        }
+    }
+
+#if MBEDTLS_VERSION_NUMBER < 0x03000000
+    mbedtls_pk_context pk1 = cert1.pk.pk_info ? cert1.pk : csr1.pk;
+    mbedtls_pk_context pk2 = cert2.pk.pk_info ? cert2.pk : csr2.pk;
+#else
+    mbedtls_pk_context pk1 = cert1.pk_raw.p ? cert1.pk : csr1.pk;
+    mbedtls_pk_context pk2 = cert2.pk_raw.p ? cert2.pk : csr2.pk;
+#endif
+
+    if(!mbedtls_pk_rsa(pk1) || !mbedtls_pk_rsa(pk2)) {
+        retval = UA_STATUSCODE_BADINTERNALERROR;
+        goto cleanup;
+    }
+
+    if(!mbedtls_pk_can_do(&pk1, MBEDTLS_PK_RSA) &&
+       !mbedtls_pk_can_do(&pk2, MBEDTLS_PK_RSA)) {
+        retval = UA_STATUSCODE_BADINTERNALERROR;
+        goto cleanup;
+    }
+
+    /* Initialize the hashing context with SHA-256 */
+    const mbedtls_md_info_t *md_info = mbedtls_md_info_from_type(MBEDTLS_MD_SHA256);
+    if(!md_info) {
+        retval = UA_STATUSCODE_BADOUTOFMEMORY;
+        goto cleanup;
+    }
```

Comment:
Where is this context used?

Reply:
You're right. It's a leftover and will be removed.